api/proposal/proposal.go

```diff
@@ -109,3 +139,62 @@ func getAllProposalHandler(res http.ResponseWriter, req *http.Request) {
 
 	httputil.WriteData(res, http.StatusOK, docProps)
 }
+
+// @Summary		Approve a proposal
+// @Description	Approve a proposal given its id
+// @Tags			answer
+// @Produce		json
+// @Success		200	{object}	models.Question
+// @Failure		400	{object}	httputil.ApiError
+// @Router			/proposals/{id}/approve [post]
+func ApproveProposalHandler(res http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		httputil.WriteError(res, http.StatusMethodNotAllowed, "invalid method")
+		return
+	}
+
+	if !middleware.GetAdmin(req) {
+		httputil.WriteError(res, http.StatusForbidden, "you are not admin")
+		return
+	}
+
+	rawID := muxie.GetParam(res, "id")
+	proposalID, err := strconv.Atoi(rawID)
+	if err != nil {
+		httputil.WriteError(res, http.StatusBadRequest, "invalid id")
+		return
+	}
+
+	db := util.GetDb()
+	var proposal Proposal
+	var question models.Question
+
+	err = db.Transaction(func(tx *gorm.DB) error {
+		if err := tx.Clauses(clause.Returning{}).Delete(&proposal, proposalID).Error; err != nil {
+			tx.Rollback()
```

Comment:
You don't need to rollback, just returning the error inside the Transaction function will make gorm rollback on its own